sale_commission/wizard/wizard_settle.py

```diff
@@ -47,6 +47,23 @@ def _get_next_period_date(self, agent, current_date):
         else:
             raise exceptions.Warning(_("Settlement period not valid."))
 
+    def _get_settlement_domain(self, agent, company, sett_from, sett_to):
+        return [
+            ('agent', '=', agent.id),
+            ('date_from', '=', sett_from),
+            ('date_to', '=', sett_to),
+            ('company_id', '=', company.id),
+            ('state', '=', 'settled')
+        ]
+
+    def _get_settlement_values(self, agent, company, sett_from, sett_to):
```

Comment:
Call it for following the same practice as Odoo core, `_prepare_settlement_vals`